test/testUtils.ts

```diff
@@ -0,0 +1,13 @@
+import t from 'tap'
+
+export async function throwsWithMessage(
+  t: Tap.Test,
+  funcUnderTest: () => {},
+  expectedMessage: string
+) {
+  try {
+    await funcUnderTest()
+  } catch (error: any) {
+    t.same(error.message, expectedMessage)
```

Comment:
Hmm. Won't this succeed if the function _doesn't_ throw as well? 🤔

Reply:
Excellent point. I could call the function twice, once to ensure the exception happens, and again to check the error message, but that feels wasteful.

I'll see if there's a built-in way to test the exact message being thrown, a deeper dive into the docs might be necessary.